scripts/install_rust.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+echo "*** Checking if Rust is already installed"
+
+if which rustup >/dev/null 2>&1; then
+    echo "Rust is already installed. Exiting."
+    exit 0
+fi
+
+echo "*** Installing Rust"
+
+if [[ "$(uname)" == "Darwin" ]]; then
+    # macOS
+    if ! which brew >/dev/null 2>&1; then
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+    fi
+    
+    brew update
+    brew install openssl cmake llvm
+    elif [[ "$(uname)" == "Linux" ]]; then
+    if [[ -f "/etc/arch-release" ]]; then
+        # Arch Linux
+        sudo pacman -Syu --noconfirm
+        sudo pacman -S --noconfirm cmake pkgconf openssl git gcc clang
+    else
+        # Ubuntu (and other Debian-based distributions)
+        sudo apt update
```

Comment:
note that usage of plain `apt` in scripts is unstable, still supposed to use `apt-get` because it has a stable CLI that won't i.e. spawn some UI sometimes
```suggestion
        sudo apt-get update
```